nix-tools/make-install-plan/MakeInstallPlan.hs

```diff
@@ -0,0 +1,128 @@
+import Distribution.Client.HashValue (HashValue, showHashValue)
+import qualified Distribution.Client.InstallPlan as InstallPlan
+import Distribution.Client.NixStyleOptions (NixStyleFlags (..), defaultNixStyleFlags, nixStyleOptions)
+import Distribution.Client.ProjectConfig
+import Distribution.Client.ProjectOrchestration
+import Distribution.Client.ProjectPlanOutput (writePlanExternalRepresentation)
+import Distribution.Client.ProjectPlanning (ElaboratedConfiguredPackage (..), rebuildInstallPlan)
+import Distribution.Client.Setup
+import Distribution.Client.Types.PackageLocation (PackageLocation (..))
+import Distribution.Client.Types.Repo (LocalRepo (..), RemoteRepo (..), Repo (..))
+import Distribution.Client.Types.SourceRepo (SourceRepositoryPackage (..))
+import Distribution.Package (pkgName)
+import Distribution.Pretty (prettyShow)
+import Distribution.Simple.Command
+import Distribution.Simple.Flag
+import qualified Distribution.Simple.Utils as Cabal
+import Distribution.Types.SourceRepo (KnownRepoType (Git), RepoType (..))
+import Distribution.Verbosity (Verbosity)
+import qualified Distribution.Verbosity as Verbosity
+import Nix.Pretty (prettyNix)
+import Prettyprinter (Doc)
+import Prettyprinter.Render.Text (hPutDoc)
+import System.Environment (getArgs)
+import System.FilePath
+import System.IO (IOMode (WriteMode), hClose, openFile)
+
+main :: IO ()
+main = do
+  args <- getArgs
+  case commandParseArgs cmdUI True args of
+    CommandHelp help -> putStrLn (help "something")
+    CommandList opts -> putStrLn $ "commandList" ++ show opts
+    CommandErrors errs -> putStrLn $ "commandErrors: " ++ show errs
+    CommandReadyToGo (mkflags, _commandParse) ->
+      let globalFlags = defaultGlobalFlags
+          flags@NixStyleFlags {configFlags} = mkflags (commandDefaultFlags cmdUI)
+          verbosity = fromFlagOrDefault Verbosity.normal (configVerbosity configFlags)
+          cliConfig = commandLineFlagsToProjectConfig globalFlags flags mempty
+       in installPlanAction verbosity cliConfig
+
+cmdUI :: CommandUI (NixStyleFlags ())
+cmdUI =
+  CommandUI
+    { commandName = "",
+      commandSynopsis = "Makes an install-plan",
+      commandUsage = ("Usage: " ++),
+      commandDescription = Nothing,
+      commandNotes = Nothing,
+      commandDefaultFlags = defaultNixStyleFlags (),
+      commandOptions = nixStyleOptions (const [])
+    }
+
+installPlanAction :: Verbosity -> ProjectConfig -> IO ()
+installPlanAction verbosity cliConfig = do
+  ProjectBaseContext {distDirLayout, cabalDirLayout, projectConfig, localPackages} <-
+    establishProjectBaseContext verbosity cliConfig OtherCommand
+
+  -- Two variants of the install plan are returned: with and without
+  -- packages from the store. That is, the "improved" plan where source
+  -- packages are replaced by pre-existing installed packages from the
+  -- store (when their ids match), and also the original elaborated plan
+  -- which uses primarily source packages.
```

Comment:
I was expecting you to tell me which one we're using and why!

Reply:
It's verbatim from cabal, I just removed the comment.